### PR TITLE
Issue77

### PR DESCRIPTION
**Pull request by twillouer**
add $type to ExpressionParser

### CODE REVIEW

File: src/main/java/com/foursquare/fongo/impl/ExpressionParser.java [hunk 1]
```diff
@@ -341,6 +352,59 @@ boolean listContainsPattern(List<Object> list, Pattern pattern) {
...
+   * Max key	127
+   */
+  boolean objectMatchesType(Object obj, int type) {
+    switch (type) {
+      case 1:
+        return obj instanceof Double || obj instanceof Float;
+      case 2:
+        return obj instanceof CharSequence;
+      case 3:
+        return obj instanceof Object; // TODO verify
```
Comment (hoffrocket) [inline src/main/java/com/foursquare/fongo/impl/ExpressionParser.java]: this should probably be DBObject

File: src/test/java/com/foursquare/fongo/FongoTest.java [hunk 2]
```diff
@@ -1132,6 +1133,70 @@ public void shouldInsertIdFirst() throws Exception {
     }
   }
 
+  // http://docs.mongodb.org/manual/reference/operator/type/
+  @Test
+  public void shouldFilterByType() throws Exception {
+    // Given
+    DBCollection collection = newCollection();
+//    DBCollection collection = new MongoClient().getDB("test").getCollection("typeFilter");
```
Comment (hoffrocket) [inline src/test/java/com/foursquare/fongo/FongoTest.java]: remove commented out code

File: src/main/java/com/foursquare/fongo/impl/ExpressionParser.java (hunk 1 above)
Comment (twillouer) [inline src/main/java/com/foursquare/fongo/impl/ExpressionParser.java]: I try and... it's really Object... (I commited the test)

File: src/test/java/com/foursquare/fongo/FongoTest.java (hunk 2 above)
Comment (twillouer) [inline src/test/java/com/foursquare/fongo/FongoTest.java]: right.. I use FongoRule who help me a lot for this :)

**Comment by twillouer** [comment]
updated

**Comment by hoffrocket** [comment]
there are merge conflicts, can you update?

**Comment by twillouer** [comment]
done

**Pull request merged**